src/db/transaction.rs

```diff
@@ -144,12 +168,18 @@ impl DbTransactionPool {
             }
         };
 
-        let (resp, db) = self.transaction_internal(check_precondition).await?;
+        let (mut resp, db) = self
+            .transaction_internal(check_precondition, request.clone())
+            .await?;
+        // match on error and return a composed HttpResponse (so we can use the tags?)
```

Comment:
Let's either a) only call `set_extra` on the error cases (which would be here as this comment describes and also below when there's a `resp.error()` OR b) call it on every single request (which is what the line 152 call is pretty much doing).

We're kinda doing both here. a) is probably nicer but b) might be ok, it's pretty cheap I think -- but in that case maybe call it at the beginning of the check_precondition closure as e.g. `extract_resource` could potentially error out before it was called

Reply:
Adding some commentary around the `set_extra` calls.